lib/cmsis_rtos_v1/cmsis_timer.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel_structs.h>
+#include <cmsis_os.h>
+
+#define ACTIVE 1
+#define NOT_ACTIVE 0
+
+static void zephyr_timer_wrapper(struct k_timer *timer);
+
+struct timer_obj {
+	struct k_timer ztimer;
+	os_timer_type type;
+	u32_t status;
+	void (*callback_function)(void const *argument);
+	void *arg;
+};
+
+K_MEM_SLAB_DEFINE(cmsis_timer_slab, sizeof(struct timer_obj),
+		  CONFIG_MAX_TIMER_COUNT, 4);
+
+static void zephyr_timer_wrapper(struct k_timer *ztimer)
+{
+	struct timer_obj *timer;
+
+	timer = (struct timer_obj *)ztimer;
+
+	(timer->callback_function)(timer->arg);
```

Comment:
Use CONTAINER_OF() here.  Yeah, I see you put the k_timer at the beginning of the struct.  That's a clever footgun.  Do it right :)

Reply:
Will do.